event/src/test/java/com/bazaarvoice/emodb/event/db/astyanax/DefaultSlabAllocatorTest.java

```diff
@@ -0,0 +1,165 @@
+            _log.info("SUCCESS: allocation has " + allocation.getLeft() + " slots, " + (Constants.MAX_SLAB_SIZE/2) + " expected when " + 0 + " slots already used and " + (Constants.MAX_SLAB_SIZE/2) + " requested");
+        } catch (Exception e) {
+            _log.error("ERROR: " + e.getClass().getName() + " thrown when " + (Constants.MAX_SLAB_SIZE/2) + " events allocated");
+            assert(false);
+        }
+    }
+
+    @Test
+    public void computeAllocationTotalMaxEvents() {
+        List<Integer> sizes = new ArrayList<Integer>();
+        for (int i=0; i<Constants.MAX_SLAB_SIZE/2; i++) {
+            sizes.add(64);
+        }
+        try {
+            Pair<Integer, Integer> allocation = DefaultSlabAllocator.defaultAllocationCount(Constants.MAX_SLAB_SIZE/2,0, Iterators.peekingIterator(sizes.iterator()));
+            assertTrue(allocation.getLeft() == Constants.MAX_SLAB_SIZE/2, "ERROR: allocation has " + allocation.getLeft() + " slots, " + (Constants.MAX_SLAB_SIZE/2) + " expected");
+            _log.info("SUCCESS: allocation has " + allocation.getLeft() + " slots, " + (Constants.MAX_SLAB_SIZE/2) + " expected when " + (Constants.MAX_SLAB_SIZE/2) + " slots already used and " + Constants.MAX_SLAB_SIZE/2 + " requested");
+        } catch (Exception e) {
+            _log.error("ERROR: " + e.getClass().getName() + " thrown when " + (Constants.MAX_SLAB_SIZE/2) + " events allocated");
+            assert(false);
+        }
+    }
+
+    @Test
+    public void computeAllocationTotalGreaterThanMaxEvents() {
+        List<Integer> sizes = new ArrayList<Integer>();
+        for (int i=0; i<Constants.MAX_SLAB_SIZE; i++) {
+            sizes.add(64);
+        }
+        try {
+            Pair<Integer, Integer> allocation = DefaultSlabAllocator.defaultAllocationCount(Constants.MAX_SLAB_SIZE/2,0, Iterators.peekingIterator(sizes.iterator()));
+            assertTrue(allocation.getLeft() == Constants.MAX_SLAB_SIZE/2, "ERROR: allocation has " + allocation.getLeft() + " slots, " + (Constants.MAX_SLAB_SIZE/2) + " expected");
+            _log.info("SUCCESS: allocation has " + allocation.getLeft() + " slots, " + (Constants.MAX_SLAB_SIZE/2) + " expected when " + (Constants.MAX_SLAB_SIZE/2) + " slots already used and " + Constants.MAX_SLAB_SIZE + " requested");
+        } catch (Exception e) {
+            _log.error("ERROR: " + e.getClass().getName() + " thrown when " + (Constants.MAX_SLAB_SIZE/2) + " events allocated");
+            assert(false);
+        }
+    }
+
+    @Test
+    public void computeAllocationTotalGreaterThanMaxEventsWithOversizeEvent() {
+        List<Integer> sizes = new ArrayList<Integer>();
+        for (int i=0; i<Constants.MAX_SLAB_SIZE-1; i++) {
+            sizes.add(64);
+        }
+        sizes.add(Constants.MAX_EVENT_SIZE_IN_BYTES+1);
+        try {
+            Pair<Integer, Integer> allocation = DefaultSlabAllocator.defaultAllocationCount(Constants.MAX_SLAB_SIZE/2,0, Iterators.peekingIterator(sizes.iterator()));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "allocation". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.qa.bazaarvoice.com/coding_rules#rule_key=squid%3AS1854)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused "allocation" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.qa.bazaarvoice.com/coding_rules#rule_key=squid%3AS1481)